internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go

```diff
@@ -1506,13 +1507,19 @@ func Upgrade(_ runtime.Sequence, data any) (runtime.TaskExecutionFunc, string) {
 
 		logger.Printf("performing upgrade via %q", in.GetImage())
 
+		rgs, err := safe.ReaderGetByID[*crires.RegistriesSpec](ctx, r.State().V1Alpha2().Resources(), crires.RegistriesID)
```

Comment:
this certainly has a race condition (not here, but more on the install sequence)

a proper way would be to block for the config to be available

```suggestion
		rgs, err := safe.StateWaitFor[*crires.RegistriesSpec](ctx, r.State().V1Alpha2().Resources(), crires.RegistriesID, state.WithEventTypes(state.Created, state.Updated))
```

Reply:
Done!